modules/telco-upgrade-upgrading-the-olm-operators.adoc

```diff
@@ -0,0 +1,106 @@
+$ oc patch installplan -n metallb-system install-nwjnh --type merge --patch \
+'{"spec":{"approved":true}}'
+----
++
+.Example output
+[source,terminal]
+----
+installplan.operators.coreos.com/install-nwjnh patched
+----
+
+. Monitor the namespace:
++
+[source,terminal]
+----
+$ oc get all -n metallb-system
+----
++
+.Example output
+[source,terminal]
+----
+NAME                                                       READY   STATUS              RESTARTS   AGE
+pod/metallb-operator-controller-manager-69b5f884c-8bp22    0/1     ContainerCreating   0          4s
+pod/metallb-operator-controller-manager-77895bdb46-bqjdx   1/1     Running             0          4m1s
+pod/metallb-operator-webhook-server-5d9b968896-vnbhk       0/1     ContainerCreating   0          4s
+pod/metallb-operator-webhook-server-d76f9c6c8-57r4w        1/1     Running             0          4m1s
+
+#...
+
+NAME                                                             DESIRED   CURRENT   READY   AGE
+replicaset.apps/metallb-operator-controller-manager-69b5f884c    1         1         0       4s
+replicaset.apps/metallb-operator-controller-manager-77895bdb46   1         1         1       4m1s
+replicaset.apps/metallb-operator-controller-manager-99b76f88     0         0         0       4m40s
+replicaset.apps/metallb-operator-webhook-server-5d9b968896       1         1         0       4s
+replicaset.apps/metallb-operator-webhook-server-6f7dbfdb88       0         0         0       4m40s
+replicaset.apps/metallb-operator-webhook-server-d76f9c6c8        1         1         1       4m1s
+----
++
+When the update is complete, the updated namespace resources look like this:
++
+[source,terminal]
+----
+NAME                                                      READY   STATUS    RESTARTS   AGE
+pod/metallb-operator-controller-manager-69b5f884c-8bp22   1/1     Running   0          25s
+pod/metallb-operator-webhook-server-5d9b968896-vnbhk      1/1     Running   0          25s
+
+#...
+
+NAME                                                             DESIRED   CURRENT   READY   AGE
+replicaset.apps/metallb-operator-controller-manager-69b5f884c    1         1         1       25s
+replicaset.apps/metallb-operator-controller-manager-77895bdb46   0         0         0       4m22s
+replicaset.apps/metallb-operator-webhook-server-5d9b968896       1         1         1       25s
+replicaset.apps/metallb-operator-webhook-server-d76f9c6c8        0         0         0       4m22s
+----
+
+.Verification
+* Verify that the Operators do not need to be updated for a second time:
++
+[source,terminal]
+----
+$ oc get installplan -A | grep -E 'APPROVED|false'
+----
++
+There should be no output returned.
++
+[NOTE]
+====
+Sometimes you need to approve an update twice because some Operators have interim z-stream release versions that need to be installed before the final version.
+====
+
+[NOTE]
+====
+If you are performing a y-stream or z-stream upgrade than you can skip to "Upgrading the worker nodes".
```

Comment:
As opposed to what, an EUS /Control Plane Only update?

This is a bit confusing. I can't quite follow. 

So if it's a y-stream or z-stream, then they can skip "Performing the second y-stream update", "Acknowledging the y-stream release upgrade", "Starting the y-stream control plane upgrade", "Monitoring the second part of a <y+1> cluster upgrade", and "Upgrading all the OLM Operators"?

If so, then I'm confused by the indentation. It seems like all of that should be indented under it's only section for EUS updates. But right now, only the first two are indented under updating operators. Is something off here?

Reply:
I've completely reworked this. There are now 3 discrete sections:
1. Performing a Control Plane Only updates
2. Performing a y-stream updates
3. Performing a z-stream updates